src/planscape/climate_foresight/views.py

```diff
@@ -69,6 +113,90 @@ def datalayers(self, request):
         serializer = BrowseDataLayerSerializer(datalayers, many=True)
         return Response(serializer.data)
 
+    @action(detail=True, methods=["post"])
+    def run_analysis(self, request, pk=None):
+        """
+        Start the full Climate Foresight analysis pipeline.
+
+        This endpoint kicks off the entire workflow:
+        1. Normalize all input layers
+        2. Rollup pillars
+        3. Rollup landscape (current + future)
+        4. Run PROMOTe analysis (MPAT outputs)
+
+        The run must be in DRAFT status and have:
+        - All input layers with favor_high set
+        - All input layers assigned to pillars
+
+        Returns a summary of what was started.
+        """
+        run = self.get_object()
+
+        try:
+            result = start_climate_foresight_analysis(run.id)
+            return Response(result, status=status.HTTP_200_OK)
+        except ValueError as e:
+            raise ValidationError(str(e))
+
+    @action(detail=True, methods=["post"])
+    def trigger_next_steps(self, request, pk=None):
+        """
+        Manually trigger the next ready steps in the analysis pipeline.
+
+        This is useful for debugging or when automatic progression fails.
+        It checks what's ready and triggers:
+        - Pillar rollups (if all input layers normalized)
+        - Landscape rollup (if all pillars completed)
+        - PROMOTe analysis (if landscape completed)
+        - Mark run as DONE (if PROMOTe completed)
+
+        Returns a summary of what was triggered.
+        """
+        run = self.get_object()
+
+        results = {
+            "run_id": run.id,
+            "pillar_rollups": trigger_pillar_rollups_if_ready(run.id),
+            "landscape_rollup": trigger_landscape_rollup_if_ready(run.id),
+            "promote": trigger_promote_if_ready(run.id),
+            "completion_check": check_run_completion(run.id),
+        }
+
+        return Response(results, status=status.HTTP_200_OK)
+
+    @extend_schema(
+        description="Download all Climate Foresight outputs as a zipped archive of GeoTIFFs.",
+        responses={
+            200: OpenApiTypes.BINARY,
+            400: BaseErrorMessageSerializer,
+            404: BaseErrorMessageSerializer,
+        },
+    )
+    @action(detail=True, methods=["get"], filterset_class=None)
+    def download(self, request, pk=None):
+        """
+        Export and download all Climate Foresight run outputs.
+
+        Returns a zip file containing GeoTIFF rasters for:
+        - MPAT outputs (matrix, strength, individual strategies)
+        - Pillar rollups
+        - Landscape rollups (current and future)
+        - Normalized input layers
+
+        The run must be in 'done' status.
+        """
+        run = self.get_object()
+
+        try:
+            output_path = export_geopackage(run.id)
```

Comment:
For large areas this can take QUITE A WHILE. Test and see. Most likely this won't work for all cases.